lib/SILOptimizer/Analysis/ConcreteTypeAnalysis.cpp

```diff
@@ -0,0 +1,98 @@
+//===--- ConcreteTypeAnalysis.cpp - Protocol to Class inheritance ---------===//
```

Comment:
Word "Class" should not appear here, and maybe rename this to "ProtocolConfomanceAnalysis"

in general we use "X conforms to Y" instead of "X implements Y"